src/components/App/App.js

```diff
@@ -0,0 +1,165 @@
+import './App.css';
+
+import React, { useState } from 'react';
+import { Base64 } from 'js-base64';
+import jsQR from 'jsqr';
+import pako from 'pako';
+import HealthCardDisplay from 'components/HealthCardDisplay';
+
+function App() {
+  const [isScanning, setIsScanning] = useState(false);
+  const [qrData, setQrData] = useState(null);
+
+  const decodeQr = (qrString) => {
+    const sliceIndex = qrString.lastIndexOf('/');
+    const rawPayload = qrString.slice(sliceIndex + 1);
+    const encodingChars = rawPayload.match(/\d\d/g);
+    const jwsString = encodingChars.map((charPair) => String.fromCharCode(+charPair + 45)).join('');
+    const dataString = jwsString.split('.')[1];
+    const decodedPayload = Base64.toUint8Array(dataString);
+    const inflatedPayload = pako.inflateRaw(decodedPayload);
+    const payload = new TextDecoder().decode(inflatedPayload);
+    return payload;
+  };
+
+  const extractPatientName = (patient) => {
+    const nameElement = patient.name[0];
+
+    if (nameElement.text) {
+      return nameElement.text;
+    }
+
+    const prefix = nameElement.prefix ? nameElement.prefix.join(' ') : '';
+    const given = nameElement.given ? nameElement.given.join(' ') : '';
+    const family = nameElement.family ? nameElement.family : '';
+    const suffix = nameElement.suffix ? nameElement.suffix.join(' ') : '';
+
+    const name = [prefix, given, family, suffix].join(' ');
```

Comment:
It looks like the way the name sub-strings are concatenated together adds leading and trailing whitespace in the returned name string when there are no prefixes or suffixes. When I log the patient to the console, I see `name: " BethesdaAIRA Delvene WeilAIRA "`. I'm not sure if this matters a whole lot right now, but the extra space could become more apparent once the UI is updated.

Reply:
Good catch, I removed the leading and trailing whitespace.